src/main/java/com/hivemq/HiveMQServer.java

```diff
@@ -134,16 +136,18 @@ public static void main(final @NotNull String[] args) throws Exception {
         if (ShutdownHooks.SHUTTING_DOWN.get()) {
             return;
         }
-
-        if (migrations.size() > 0) {
-            log.info("Persistence types has been changed, migrating persistent data.");
-            for (final MigrationUnit migrationUnit : migrations.keySet()) {
-                log.debug("{} needs to be migrated.", StringUtils.capitalize(migrationUnit.toString()));
+        if (configService.persistenceConfigurationService().getMode() == PersistenceMode.IN_MEMORY) {
```

Comment:
shouldn't this if be `!=`

Reply:
Good catch, I can't really unit test the main. We may have to port the "old" ITs to the CE soon.